### ISSUE TITLE
Can't publish proposal 

### ISSUE THREAD

**Issue by Quentinchampenois** [mainio/decidim-module-simple_proposal#1]
Hello, proposal form validations are failing with `require_scope` and `require_category` enabled.

### Description

As a user, when I try to create a proposal I get an error message if categories and scope selection are disabled on the participatory process.

Since [display conditions in form](https://github.com/mainio/decidim-module-simple_proposal/blob/1356edd84a6e34ad86c4571f8c0145f8f2da82c6/app/views/decidim/proposals/proposals/_edit_form_fields.html.erb#L47) are different from [condition in form validations](https://github.com/mainio/decidim-module-simple_proposal/blob/1356edd84a6e34ad86c4571f8c0145f8f2da82c6/app/form/concerns/decidim/simple_proposal/proposal_form_override.rb#L21), for a given proposal with no scope, user can't be able to finish the form.

### Additional context

We'd like to use it on a platform and enable it even when scopes and categories are disabled on the component, would you be interested in this enhancement of the module ?

**Comment by lahdeero** [comment]
Hello @Quentinchampenois ! Sorry for delay, I didn't get any notifications about this issue. Thanks to you I noticed that form validations were mixed.

Yeah I can see your point, that it would make more sense if display conditions would match form conditions. Maybe we should change this or allow you to open PR, what do you think @ahukkanen ? 

But for now, could you just disable category and scope requirements in initializer? Like this:
```ruby
# config/initializers/simple_proposal.rb
Decidim::SimpleProposal.configure do |config|
  config.require_category = false
  config.require_scope = false
end
```

**Comment by ahukkanen** [comment]
I think we can fix this, as we are updating the modules anyways.

I believe the best would be to use the methods already defined in the form class also in the view, to standardize their behavior, similarly as here:
https://github.com/mainio/decidim-module-simple_proposal/blob/2147b41cafc53eef0557d29f24227d9fcd903fa9/app/views/decidim/proposals/proposals/_edit_form_fields.html.erb#L35

Also, the scope count check here should be fixed:
https://github.com/mainio/decidim-module-simple_proposal/blob/1356edd84a6e34ad86c4571f8c0145f8f2da82c6/app/form/concerns/decidim/simple_proposal/proposal_form_override.rb#L22

It should be as it is currently in the view:
https://github.com/mainio/decidim-module-simple_proposal/blob/2147b41cafc53eef0557d29f24227d9fcd903fa9/app/views/decidim/proposals/proposals/_edit_form_fields.html.erb#L53

**Comment by lahdeero** [comment]
Should be fixed now!

**Comment by ahukkanen** [comment]
Fixed by 935a72fcf533a5695d3750d89e87f389c189c1a9.
